include/wobbly/geometry_traits.h

```diff
@@ -0,0 +1,370 @@
+                    {
+                        set <D> (point, v (get <D> (point)));
+                        MutateRecursive <T, Visitor, D - 1>::apply (point,
+                                                                    std::forward <Visitor> (v));
+                    }
+                };
+
+                template <typename T, typename Visitor>
+                struct MutateRecursive <T, Visitor, 0>
+                {
+                    static void apply (T &point, Visitor &&v)
+                    {
+                        set <0> (point, v (get <0> (point)));
+                    }
+                };
+
+                template <typename T, typename U, typename Visitor, size_t D>
+                struct VectorApplyOntoRecursive
+                {
+                    static void apply (T &dst, U const &src, Visitor &&v)
+                    {
+                        set <D> (dst, v (get <D> (dst), get <D> (src)));
+                        VectorApplyOntoRecursive <T, U, Visitor, D - 1>::apply (dst,
+                                                                                src,
+                                                                                std::forward <Visitor> (v));
+                    }
+                };
+
+                template <typename T, typename U, typename Visitor>
+                struct VectorApplyOntoRecursive <T, U, Visitor, 0>
+                {
+                    static void apply (T &dst, U const &src, Visitor &&v)
+                    {
+                        set <0> (dst, v (get<0> (dst), get <0> (src)));
+                    }
+                };
+
+                template <typename T, typename U, typename Visitor, size_t D>
+                struct ScalarApplyOntoRecursive
+                {
+                    static void apply (T &dst, U const &src, Visitor &&v)
+                    {
+                        set <D> (dst, v (get <D> (dst), src));
+                        ScalarApplyOntoRecursive <T, U, Visitor, D - 1>::apply (dst,
+                                                                                src,
+                                                                                std::forward <Visitor> (v));
+                    }
+                };
+                template <typename T, typename U, typename Visitor>
+                struct ScalarApplyOntoRecursive <T, U, Visitor, 0>
+                {
+                    static void apply (T &dst, U const &src, Visitor &&v)
+                    {
+                        set <0> (dst, v (get <0> (dst), src));
+                    }
+                };
+
+                template <typename T, typename U, typename A, typename Visitor, size_t D>
+                struct ReduceCoordinatePairsRecursive
+                {
+                    static A apply (T const &l,
+                                    U const &r,
+                                    A       &&acc,
```

Comment:
Nitpick: The alignment here differs from the rest of the file.

Reply:
Does it? Not sure what you mean here.